src/python_testing/TC_OPSTATE_2_3.py

```diff
@@ -86,32 +85,34 @@ async def test_TC_OPSTATE_2_3(self):
-
-        self.print_step(7, "Waiting for 5 seconds")
-        time.sleep(5)
-
-        self.print_step(8, "Read CountdownTime attribute")
-        countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CountdownTime)
-        logging.info("CountdownTime: %s" % (countdown_time))
-        asserts.assert_true(countdown_time != 0 or countdown_time == NullValue,
-                            "invalid CountdownTime(%s). Must be a non zero integer, or null" % countdown_time)
-        asserts.assert_equal(countdown_time, initial_countdown_time, "CountdownTime(%s) does not equal to the intial CountdownTime(%s)"
-                             % (countdown_time, initial_countdown_time))
+        if self.check_pics("OPSTATE.S.M.ST_PAUSED"):
+            self.print_step(5, "Read OperationalState attribute")
+            operational_state = await self.read_mod_attribute_expect_success(endpoint=self.endpoint,
+                                                                             attribute=attributes.OperationalState)
+            logging.info("OperationalState: %s" % (operational_state))
+            asserts.assert_equal(operational_state, Clusters.OperationalState.Enums.OperationalStateEnum.kPaused,
+                                 "OperationalState ID should be Paused(0x02)")
+
+        if self.check_pics("OPSTATE.S.A0002"):
+            self.print_step(6, "Read CountdownTime attribute")
+            initial_countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint,
+                                                                                  attribute=attributes.CountdownTime)
+            logging.info("CountdownTime: %s" % (initial_countdown_time))
+            if initial_countdown_time is not NullValue:
+                in_range = (1 <= initial_countdown_time <= 259200)
+            asserts.assert_true(initial_countdown_time is NullValue or in_range,
+                                "invalid CountdownTime(%s). Must be in between 1 and 259200, or null " % initial_countdown_time)
+
+            self.print_step(7, "Waiting for 5 seconds")
+            time.sleep(5)
+
+            self.print_step(8, "Read CountdownTime attribute")
+            countdown_time = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CountdownTime)
+            logging.info("CountdownTime: %s" % (countdown_time))
+            asserts.assert_true(countdown_time != 0 or countdown_time == NullValue,
+                                "invalid CountdownTime(%s). Must be a non zero integer, or null" % countdown_time)
+            asserts.assert_equal(countdown_time, initial_countdown_time, "CountdownTime(%s) does not equal to the intial CountdownTime(%s)"
+                                 % (countdown_time, initial_countdown_time))
 
         self.print_step(9, "Send Pause command")
```

Comment:
If `OPSTATE.S.M.ST_PAUSED` is not supported, wouldn't this step fail? 

Logically, I don't see how the resume commands can be tested if `OPSTATE.S.M.ST_PAUSED` is not supported since there is no other way to put the device in a state from which it can resume. If there is a way, the test plan needs to add a manual step saying "Manually put the DUT into a state wherein it can receive a Resume Command". This step could be on the condition that `OPSTATE.S.M.ST_PAUSED` is not supported.